libs/fairness/src/lib/Controls/ModelComparisonChart.tsx

```diff
@@ -341,15 +386,49 @@ export class ModelComparisonChart extends React.Component<
             this.props.featureBinPickerProps.selectedBinIndex,
             modelIndex,
             this.props.fairnessPickerProps.selectedFairnessKey,
-            fairnessOption.fairnessMode
+            fairnessOption.fairnessMode,
+            this.props.errorPickerProps.selectedErrorKey
           );
         });
 
       const performanceArray = (await Promise.all(performancePromises)).map(
-        (metric) => metric.global
+        (metric) => {
+          console.log(metric);
+          return metric.global;
+        }
+      );
+
+      const performanceBounds = (await Promise.all(performancePromises)).map(
```

Comment:
There are only 2 requests right? One for fairness metrics and one for performance metrics. For each performance/fairness, I have 2 map functions to get the `performanceArray` and `performanceBounds` to extract from `metric.global` and `metric.bounds` respectively. Each of these map functions await on the same promise, so it should only do 1 request right? I used 2 map functions because I don't know a better way to do this in TypeScript, but please let me know if there is one

Reply:
Ah, I got confused by having await there twice and didn't read further. Perhaps do the `await Promise.all(performancePromises)` part first, then you can reuse the result in both places.

You're right on it being 2 requests total which makes sense. 2 per metric anyway.